src/Chainweb/Pact/PactService.hs

```diff
@@ -538,9 +549,13 @@ execNewGenesisBlock
 execNewGenesisBlock miner newTrans = withDiscardedBatch $ do
     pdbenv <- restoreCheckpointer Nothing
     results <- execTransactions Nothing miner newTrans pdbenv
+    finalizeCheckpointer discard
```

Comment:
This call is new. Without it the function would leave an Block savepoint on the sqlite transaction stack.